### PR TITLE
Add API for autocompleting users

### PR DESCRIPTION
**Pull request by jwilander**
API proposal for mattermost/platform#4239

### CODE REVIEW

File: source/users.yaml [hunk 1]
```diff
@@ -412,3 +412,33 @@
...
+        - name: team_id
+          in: path
+          description: The team ID the users are in
+          required: true
+          type: string
+        - name: term
+          in: query
+          description: The term to match against username, full name or nickname
+          required: true
+          type: integer
```
Comment (enahum) [inline source/users.yaml]: I think you meant **string**?

File: source/users.yaml [hunk 2]
```diff
@@ -412,3 +412,33 @@
...
+        - name: channel_id
+          in: query
+          description: The channel ID to autocomplete for
+          required: false
+          type: integer
```
Comment (enahum) [inline source/users.yaml]: same here

File: source/users.yaml (hunk 1 above)
Comment (jwilander) [inline source/users.yaml]: Woops..

**Comment by jwilander** [comment]
@enahum fixed

**Comment by jwilander** [comment]
@enahum please re-review at your leisure

**Pull request merged**